*** a/ext/openssl/xp_ssl.c	2014-08-14 04:22:50.000000000 +0900
--- b/ext/openssl/xp_ssl.c	2018-08-28 14:50:25.000000000 +0900
***************
*** 338,346 ****
--- 338,351 ----
  			break;
  #endif
  		case STREAM_CRYPTO_METHOD_SSLv3_CLIENT:
+ #ifdef OPENSSL_NO_SSL2
+ 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "SSLv2 support is not compiled into the OpenSSL library PHP is linked against");
+ 			return -1;
+ #else
  			sslsock->is_client = 1;
  			method = SSLv3_client_method();
  			break;
+ #endif
  		case STREAM_CRYPTO_METHOD_TLS_CLIENT:
  			sslsock->is_client = 1;
  			method = TLSv1_client_method();
***************
*** 350,358 ****
--- 355,368 ----
  			method = SSLv23_server_method();
  			break;
  		case STREAM_CRYPTO_METHOD_SSLv3_SERVER:
+ #ifdef OPENSSL_NO_SSL2
+ 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "SSLv2 support is not compiled into the OpenSSL library PHP is linked against");
+ 			return -1;
+ #else
  			sslsock->is_client = 0;
  			method = SSLv3_server_method();
  			break;
+ #endif
  		case STREAM_CRYPTO_METHOD_SSLv2_SERVER:
  #ifdef OPENSSL_NO_SSL2
  			php_error_docref(NULL TSRMLS_CC, E_WARNING, "SSLv2 support is not compiled into the OpenSSL library PHP is linked against");
